modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/PersistentPageMemory.java

```diff
@@ -1089,14 +1098,24 @@ private long postWriteLockPage(long absPtr, FullPageId fullId) {
 
         // Create a buffer copy if the page is scheduled for a checkpoint.
         if (isInCheckpoint(fullId) && tempBufferPointer(absPtr) == INVALID_REL_PTR) {
-            long tmpRelPtr = checkpointPool.borrowOrAllocateFreePage(tag(fullId.pageId()));
+            long tmpRelPtr;
+
+            PagePool checkpointPool = this.checkpointPool;
+            assert checkpointPool != null;
```

Comment:
Judging by the usage, it is written once at the start, do you think someone will accidentally call it before the start? It looks like no and the assert is not necessary. What do you think?